client/src/redux/sagas/datafiles.sagas.js

```diff
@@ -826,22 +826,79 @@ export function* trashFiles(action) {
   yield call(action.payload.reloadCallback);
 }
 
-export function* trashFile(system, path, id) {
+export function* trashFile(system, path) {
   yield put({
     type: 'DATA_FILES_SET_OPERATION_STATUS_BY_KEY',
-    payload: { status: 'RUNNING', key: id, operation: 'trash' },
+    payload: { status: 'RUNNING', key: system + path, operation: 'trash' },
   });
   try {
     yield call(trashUtil, 'tapis', 'private', system, path);
 
     yield put({
       type: 'DATA_FILES_SET_OPERATION_STATUS_BY_KEY',
-      payload: { status: 'SUCCESS', key: id, operation: 'trash' },
+      payload: { status: 'SUCCESS', key: system + path, operation: 'trash' },
     });
   } catch (e) {
     yield put({
       type: 'DATA_FILES_SET_OPERATION_STATUS_BY_KEY',
-      payload: { status: 'ERROR', key: id, operation: 'trash' },
+      payload: { status: 'ERROR', key: system + path, operation: 'trash' },
+    });
+  }
+}
+
+export async function emptyUtil(api, scheme, system, path) {
+  const url = `/api/datafiles/${api}/delete/${scheme}/${system}${path}/`;
+  const method = 'PUT';
+  return fetchUtil({
+    url,
+    method,
+    body: JSON.stringify({}),
+  });
+}
+
+export function* watchEmpty() {
+  yield takeLeading('DATA_FILES_EMPTY', emptyFiles);
+}
+
+export function* emptyFiles(action) {
+  const emptyCalls = action.payload.src.map((file) => {
+    return call(emptyFile, file.system, file.path);
+  });
+  const { result } = yield race({
+    result: all(emptyCalls),
+    cancel: take('DATA_FILES_MODAL_CLOSE'),
+  });
+  yield put({
+    type: 'DATA_FILES_TOGGLE_MODAL',
+    payload: { operation: 'empty', props: {} },
+  });
+  yield put({
+    type: 'ADD_TOAST',
+    payload: {
+      message: `${result.length > 1 ? `${result.length} files` : 'File'} ${
+        !result.includes('ERR') ? 'deleted' : 'failed to delete'
+      }`,
+    },
+  });
```

Comment:
Can we wrap this in a try/catch, to handle general request errors here? In that case we want to still leave the reloadCallback at the end, outside of the try/catch, and maybe throw up a toast that there was an error deleting files in general.

Reply:
So just the toast part into a try/catch? I previously had the try/catch in there, but upon review, didn't see the need for it as we want the modal to close whether there was an error or not, just the toast language changed (as I understood from a design thread on this).

Reply:
Never mind, had myself mixed up here, mixed up try/catch with if/else. I'll give try/catch a go!